modules/hooks.nix

```diff
@@ -3225,9 +3225,26 @@ lib.escapeShellArgs (lib.concatMap (ext: [ "--ghc-opt" "-X${ext}" ]) hooks.ormol
       terraform-format =
         {
           name = "terraform-format";
-          description = "Format terraform (`.tf`) files.";
-          package = tools.terraform-fmt;
-          entry = "${hooks.terraform-format.package}/bin/terraform-fmt";
+          description = "Format Terraform (`.tf`) files.";
+          package = tools.opentofu;
+          entry =
+            let
+              terraform-fmt = pkgs.writeScriptBin "terraform-fmt" ''
+                #!/usr/bin/env bash
+
+                opentofu_or_terraform() {
```

Comment:
@totoroot Based on our discussion, I made a few changes to the existing `terraform-format`. For the time being, I feel it would be easier to alter the existing hook to work with both OpenTofu and Terraform. End users can choose whether to use OpenTofu (the default) or alter the `package` to Terraform, as shown in this example:

https://github.com/sestrella/test-pre-commit-hooks.nix/blob/main/devenv.nix

I'm planning to update the PR title and the issue accordingly.